File: bench/03_sieve/bench.py
import time
import math

def is_prime(n):
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False
    return True

limit = 100_000
t0 = time.time()
count = sum(1 for n in range(limit) if is_prime(n))
ms = (time.time() - t0) * 1000
print(f"result: {count}  time: {ms:.0f}ms")

File: bench/03_sieve/bench.rs
use std::time::Instant;

fn is_prime(n: u64) -> bool {
    if n < 2 { return false; }
    if n == 2 { return true; }
    if n % 2 == 0 { return false; }
    let mut i = 3u64;
    while i * i <= n {
        if n % i == 0 { return false; }
        i += 2;
    }
    true
}

fn main() {
    let limit = 100_000u64;
    let t0 = Instant::now();
    let count = (0..limit).filter(|&n| is_prime(n)).count();
    let ms = t0.elapsed().as_millis();
    println!("result: {}  time: {}ms", count, ms);
}
